core/storage/exploration/gae_models.py

```diff
@@ -1113,3 +1135,121 @@ def get_export_policy(cls) -> Dict[str, base_models.EXPORT_POLICY]:
             'contributors_summary': base_models.EXPORT_POLICY.NOT_APPLICABLE,
             'version': base_models.EXPORT_POLICY.NOT_APPLICABLE
         })
+
+
+class ExplorationVersionHistoryModel(base_models.BaseModel):
+    """Version history model for an oppia exploration.
+
+    Version history means some information about the previous commit on each
+    state and the exploration metadata at a particular version of the
+    exploration. The information about each state comprises of the version
+    number of the exploration on which the state was previously edited,
+    the name of the state at the previous version and the id of the user who
+    committed those changes. For metadata, the information comprises of the
+    version number of the explortion on which the metadata was previously
+    edited and the id of the user who committed those changes.
+
+    A new instance of this model is created each time a new exploration
+    is created or some changes are saved in an exploration.
+
+    The id of the model is generated as follows:
+    {exploration_id}-{exploration_version}
+    """
+
+    # The id of the corresponding exploration.
+    exploration_id = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The version of the corresponding exploration.
+    exploration_version = datastore_services.IntegerProperty(
+        required=True, indexed=True)
+    # The details of the previous commit on each state at a particular
+    # version of the exploration. The json structure will look like the
+    # following:
+    # {
+    #   [state_name: str]: {
+    #     "previously_edited_in_version": int,
+    #     "state_name_in_previous_version": str,
+    #     "committer_id": str
+    #   }
+    # }
+    # The json object can have multiple keys in this case depending on
+    # the number of states.
+    state_version_history = datastore_services.JsonProperty(
+        default={}, indexed=False)
+    # The details of the previous commit on the exploration metadata at a
+    # particular version of the exploration. The json structure will look
+    # like the following:
+    # {
+    #   "previously_edited_in_version": str,
+    #   "committer_id": str.
+    # }
+    # The json object will have only two keys in this case.
```

Comment:
If the keys are fixed, then just store them as two separate fields (metadata_last_edited_version_number, metadata_last_edited_committer_id)?

Reply:
Done.